pandas/tests/frame/test_reductions.py

```diff
@@ -1763,3 +1765,141 @@ def test_prod_sum_min_count_mixed_object():
+        ],
+    )
+    result = df.max(axis=1)
+    tm.assert_series_equal(result, expected)
+
+
+def test_min_max_timestamp_timezone_nat():
+    # GH#44196
+    rng_with_tz = pd.date_range(
+        start="2021-10-01T12:00:00+02:00", end="2021-10-02T12:00:00+02:00", freq="4H"
+    )
+    df_with_tz = DataFrame(
+        data={"A": rng_with_tz, "B": rng_with_tz + pd.Timedelta(minutes=20)}
+    )
+    df_with_tz.iloc[2, 1] = pd.NaT
+
+    result = df_with_tz.max(axis=1)
+    expected = pd.Series(
+        [
+            pd.Timestamp("2021-10-01T12:20:00+02:00"),
+            pd.Timestamp("2021-10-01T16:20:00+02:00"),
+            pd.Timestamp("2021-10-01T20:00:00+02:00"),
+            pd.Timestamp("2021-10-02T00:20:00+02:00"),
+            pd.Timestamp("2021-10-02T04:20:00+02:00"),
+            pd.Timestamp("2021-10-02T08:20:00+02:00"),
+            pd.Timestamp("2021-10-02T12:20:00+02:00"),
+        ]
+    )
+    tm.assert_series_equal(result, expected)
+
+
+def test_timezone_min_max_both_axis():
+    rng_with_tz = pd.date_range(
+        start="2021-10-01T12:00:00+02:00", end="2021-10-02T12:00:00+02:00", freq="4H"
+    )
+    df_with_tz = DataFrame(
+        data={"A": rng_with_tz, "B": rng_with_tz + pd.Timedelta(minutes=20)}
+    )
+    df_with_tz.iloc[2, 1] = pd.NaT
+
+    result = df_with_tz.max(axis=1)
+    expected = df_with_tz.T.max(axis=0)
+
+    tm.assert_series_equal(result, expected)
+
+    result = df_with_tz.min(axis=1)
+    expected = df_with_tz.T.min(axis=0)
+
+    tm.assert_series_equal(result, expected)
+
+
+def test_min_max_timedelta64_nat():
+    df = DataFrame(
+        [
+            [Timedelta(minutes=20), Timedelta(days=2), Timedelta(seconds=3)],
+            [Timedelta(minutes=2, seconds=2), Timedelta(days=2, minutes=30), pd.NaT],
+        ]
+    )
+    expected = pd.Series(
+        [Timedelta(minutes=2, seconds=2), Timedelta(days=2), Timedelta(seconds=3)]
+    )
+    result = df.min(axis=0)
+    tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(df.min(axis=0), df.T.min(axis=1))
+
+    expected = pd.Series([Timedelta(seconds=3), Timedelta(minutes=2, seconds=2)])
+    result = df.min(axis=1)
+    tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(df.min(axis=1), df.T.min(axis=0))
+
+    expected = pd.Series(
+        [Timedelta(minutes=20), Timedelta(days=2, minutes=30), Timedelta(seconds=3)]
+    )
+    result = df.max(axis=0)
+    tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(df.max(axis=0), df.T.max(axis=1))
+
+    expected = pd.Series([Timedelta(days=2), Timedelta(days=2, minutes=30)])
+    result = df.max(axis=1)
+    tm.assert_series_equal(result, expected)
+    tm.assert_series_equal(df.max(axis=1), df.T.max(axis=0))
+
+
+def test_min_max_perioddtype_nat():
+    df = DataFrame(
+        [
+            [PeriodDtype(freq="20m"), PeriodDtype(freq="1h"), PeriodDtype(freq="1d")],
+            [PeriodDtype(freq="25m"), PeriodDtype(freq="2h"), pd.NaT],
```

Comment:
Try period_range, or even just take the frame from the dt64tz case and to a .to_period("D") on it